lms/djangoapps/learner_dashboard/programs.py

```diff
@@ -147,7 +147,15 @@ def render_to_fragment(self, request, program_uuid, **kwargs):  # lint-amnesty,
         }
         program_discussion_lti = ProgramDiscussionLTI(program_uuid, request)
         program_live_lti = ProgramLiveLTI(program_uuid, request)
-        discussion_or_live_configured = program_discussion_lti.is_configured or program_live_lti.is_configured
+
+        def program_tab_view_enabled() -> bool:
+            return program_tab_view_is_enabled() and (
+                program_discussion_lti.is_configured or
+                program_live_lti.is_configured or
+                industry_pathways or
+                credit_pathways
+            )
+
```

Comment:
We are replacing the logic for tab_view_enabled that means we no longer need to check the waffleflag from now on. that is 
```
        def program_tab_view_enabled() -> bool:
            return  program_discussion_lti.is_configured or
                program_live_lti.is_configured or
                industry_pathways or
                credit_pathways
            )
```

as this is an `OR` condition, I'd prioritize `industry_pathways` or `credit_pathways` before any other instance because the `is_configured` method is going to make a DB call to return True or False. As a program has configured either pathway (local variable) we wouldn't need to make an unnessassry DB call here. It will give us a slight edge over performance. 


see the implimentation of `is_configured` which eventually calls => `ProgramLiveConfiguration.get(self.program_uuid)`